src/toc.ts

```diff
@@ -305,21 +320,25 @@ function onWillSave(e: TextDocumentWillSaveEvent) {
     }
 }
 
-function loadTocConfig() {
-    let tocSectionCfg = workspace.getConfiguration('markdown.extension.toc');
-    let tocLevels = tocSectionCfg.get<string>('levels');
+/**
+ * Updates `tocConfig` and `docConfig`.
+ * @param editor The editor, from which we detect `docConfig`.
+ */
+function loadTocConfig(editor: vscode.TextEditor): void {
+    const tocSectionCfg = vscode.workspace.getConfiguration('markdown.extension.toc');
+    const tocLevels = tocSectionCfg.get<string>('levels')!;
     let matches;
     if (matches = tocLevels.match(/^([1-6])\.\.([1-6])$/)) {
         tocConfig.startDepth = Number(matches[1]);
         tocConfig.endDepth = Number(matches[2]);
     }
-    tocConfig.orderedList = tocSectionCfg.get<boolean>('orderedList');
-    tocConfig.listMarker = tocSectionCfg.get<string>('unorderedList.marker');
-    tocConfig.plaintext = tocSectionCfg.get<boolean>('plaintext');
-    tocConfig.updateOnSave = tocSectionCfg.get<boolean>('updateOnSave');
+    tocConfig.orderedList = tocSectionCfg.get<boolean>('orderedList')!;
+    tocConfig.listMarker = tocSectionCfg.get<string>('unorderedList.marker')!;
+    tocConfig.plaintext = tocSectionCfg.get<boolean>('plaintext')!;
+    tocConfig.updateOnSave = tocSectionCfg.get<boolean>('updateOnSave')!;
```

Comment:
I think it is okay to use `!` to suppress the compiler warnings. But maybe we can use `get(key, defaultValue)`?

Reply:
But that may increase **maintenance burden**:

If we decide to change the default value of a setting someday, we'll have to search the whole codebase.

---

I prefer to establish a **convention:**

Always set default value in `package.json`.

Reply:
Then let's go for `!` 👌